lib/vagrant-libvirt/action/start_domain.rb

```diff
@@ -178,16 +178,21 @@ def call(env)
               end
 
               # TPM
-              if config.tpm_path
+              if config.tpm_path || config.tpm_version
                 raise Errors::FogCreateServerError, 'The TPM Path must be fully qualified' unless config.tpm_path[0].chr == '/'
```

Comment:
This needs to be changed to:

```suggestion
                if config.tpm_path
                  raise Errors::FogCreateServerError, 'The TPM Path must be fully qualified' unless config.tpm_path[0].chr == '/'
                end
```

Otherwise this always errors with:

```
Error when updating domain settings: undefined method `[]' for nil:NilClass
```

When I use an emulated TPM:

```ruby
Vagrant.configure(2) do |config|
  config.vm.box = 'ubuntu-20.04-amd64'
  config.vm.provider 'libvirt' do |lv, config|
    lv.tpm_model = 'tpm-tis'
    lv.tpm_type = 'emulator'
    lv.tpm_version = '2.0'
...
```

Reply:
I'll have to dig back in and take a look. It's long since been paged out. :joy:

I think I was initially hoping to be able to use qemu:///session and you'd specify the path that you wanted swtpm to create the socket but it didn't work for some reason that I can't recall. Looks like the ability to run as any user was just added in a release from 4 days ago?

Reply:
Either way, my logic is wrong there. :facepalm:

Reply:
I didn't try using `tpm_path` because my machine does not have a TPM. And even if it had one, it seems we can only have a single VM using the real TPM in 'passthru' mode, which make that mode of operation kinda useless for me.

I really just want to play with the emulated TPM. And this patch make that a possibility! :-)